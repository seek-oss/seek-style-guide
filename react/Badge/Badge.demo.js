import { Badge } from 'seek-style-guide/react';

export default {
  route: '/badge',
  title: 'Badge',
  component: Badge,
  initialProps: {
    children: '13 new'
  },
  options: [
    {
      label: 'Tone',
      type: 'radio',
      states: [
        {
          label: '',
          transformProps: props => ({
            ...props
          })
        },
        {
          label: 'Accent',
          transformProps: props => ({
            ...props,
            tone: 'accent'
          })
        },
        {
          label: 'Critical',
          transformProps: props => ({
            ...props,
            tone: 'critical'
          })
        },
        {
          label: 'Info',
          transformProps: props => ({
            ...props,
            tone: 'info'
          })
        },
        {
          label: 'Positive',
          transformProps: props => ({
            ...props,
            tone: 'positive'
          })
        },
        {
          label: 'Neutral',
          transformProps: props => ({
            ...props,
            tone: 'neutral'
          })
        },
        {
          label: 'Secondary',
          transformProps: props => ({
            ...props,
            tone: 'secondary'
          })
        }
      ]
    },
    {
      label: 'Levels',
      type: 'checkbox',
      states: [
        {
          label: 'Strong',
          transformProps: props => ({
            ...props,
            strong: true
          })
        }
      ]
    }
  ]
};
