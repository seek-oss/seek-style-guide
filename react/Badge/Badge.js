// @flow

import React, { PureComponent } from 'react';
import classnames from 'classnames';
import omit from 'lodash/omit';

import styles from './Badge.less';
import { TONE } from '../private/tone';

type Tone =
  | 'accent'
  | typeof TONE.CRITICAL
  | typeof TONE.INFO
  | typeof TONE.NEUTRAL
  | typeof TONE.POSITIVE
  | 'secondary';

type Props = {
  children: React$Node,
  tone: Tone,
  strong?: boolean
};

export default class Badge extends PureComponent<Props> {
  static displayName = 'Badge';

  static defaultProps = {
    tone: TONE.NEUTRAL
  };

  props: Props;

  render() {
    const { children, tone, strong, ...restProps } = this.props;

    return (
      <div
        className={classnames({
          [styles.root]: true,
          [styles[tone]]: tone,
          [styles.strong]: strong
        })}
        {...omit(restProps, 'className')}
      >
        {children}
      </div>
    );
  }
}
