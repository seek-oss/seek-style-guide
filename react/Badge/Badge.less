@import (reference) "~seek-style-guide/theme";

@height: @row-height * @small-type-row-span;

.__strong(@color) {
  background-color: @color;
  color: @sk-white;
}

.root {
  .rawText(1.2);
  font-weight: @sk-medium;
  height: @height;
  display: inline-flex;
  align-items: center;
  border-radius: 3px;
  padding: 0 10px;

  background-color: fade(@sk-black, 10%);
  color: @sk-black;

  &.strong {
    .__strong(@sk-black);
  }
}

.info {
  background-color: fade(@sk-info, 10%);
  color: @sk-info-dark;
  &.strong {
    .__strong(@sk-info);
  }
}

.accent {
  background-color: fade(@sk-blue-lighter, 10%);
  color: @sk-blue-lighter;
  &.strong {
    .__strong(@sk-blue-lighter);
  }
}

.critical {
  background-color: fade(@sk-critical, 10%);
  color: @sk-critical-dark;
  &.strong {
    .__strong(@sk-critical);
  }
}

.secondary {
  background-color: fade(@sk-mid-gray-dark, 10%);
  color: @sk-charcoal;
  &.strong {
    .__strong(@sk-mid-gray-dark);
  }
}

.neutral {
  background-color: fade(@sk-black, 10%);
  color: @sk-black;
  &.strong {
    .__strong(@sk-black);
  }
}

.positive {
  background-color: fade(@sk-positive, 10%);
  color: @sk-positive-dark;
  &.strong {
    .__strong(@sk-positive);
  }
}
