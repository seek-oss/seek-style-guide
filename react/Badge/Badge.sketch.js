import React from 'react';
import Badge from './Badge';

export const symbols = {
  'Badge/Accent/1. Strong': (
    <Badge tone="accent" strong>
      Badge
    </Badge>
  ),
  'Badge/Accent/2. Regular': <Badge tone="accent">Badge</Badge>,
  'Badge/Critical/1. Strong': (
    <Badge tone="critical" strong>
      Badge
    </Badge>
  ),
  'Badge/Critical/2. Regular': <Badge tone="critical">Badge</Badge>,
  'Badge/Info/1. Strong': (
    <Badge tone="info" strong>
      Badge
    </Badge>
  ),
  'Badge/Info/2. Regular': <Badge tone="info">Badge</Badge>,
  'Badge/Positive/1. Strong': (
    <Badge tone="positive" strong>
      Badge
    </Badge>
  ),
  'Badge/Positive/2. Regular': <Badge tone="positive">Badge</Badge>,
  'Badge/Neutral/1. Strong': (
    <Badge tone="neutral" strong>
      Badge
    </Badge>
  ),
  'Badge/Neutral/2. Regular': <Badge tone="neutral">Badge</Badge>,
  'Badge/Secondary/1. Strong': (
    <Badge tone="secondary" strong>
      Badge
    </Badge>
  ),
  'Badge/Secondary/2. Regular': <Badge tone="secondary">Badge</Badge>
};
