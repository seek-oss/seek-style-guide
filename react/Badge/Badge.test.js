import React from 'react';
import { shallow } from 'enzyme';
import Badge from './Badge';

const renderBadge = props => shallow(<Badge {...props}>2 new</Badge>);

test('should render Badge', () => {
  const badge = renderBadge();
  expect(badge).toMatchSnapshot();
});

test('should render accent Badge', () => {
  const badge = renderBadge({ tone: 'accent' });
  expect(badge).toMatchSnapshot();
});

test('should ignore external className', () => {
  const badge = renderBadge({ className: 'foo' });
  expect(badge).toMatchSnapshot();
});

test('should render critical Badge', () => {
  const badge = renderBadge({ tone: 'critical' });

  expect(badge).toMatchSnapshot();
});

test('should render info Badge', () => {
  const badge = renderBadge({ tone: 'info' });

  expect(badge).toMatchSnapshot();
});

test('should render neutral Badge', () => {
  const badge = renderBadge({ tone: 'neutral' });
  expect(badge).toMatchSnapshot();
});

test('should render secondary Badge', () => {
  const badge = renderBadge({ tone: 'secondary' });
  expect(badge).toMatchSnapshot();
});

test('should render a strong positive Badge', () => {
  const badge = renderBadge({ tone: 'secondary', strong: true });
  expect(badge).toMatchSnapshot();
});
