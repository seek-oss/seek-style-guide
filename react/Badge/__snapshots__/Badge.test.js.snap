// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should ignore external className 1`] = `
<div
  className="Badge__root Badge__neutral"
>
  2 new
</div>
`;

exports[`should render Badge 1`] = `
<div
  className="Badge__root Badge__neutral"
>
  2 new
</div>
`;

exports[`should render a strong positive Badge 1`] = `
<div
  className="Badge__root Badge__secondary Badge__strong"
>
  2 new
</div>
`;

exports[`should render accent Badge 1`] = `
<div
  className="Badge__root Badge__accent"
>
  2 new
</div>
`;

exports[`should render critical Badge 1`] = `
<div
  className="Badge__root Badge__critical"
>
  2 new
</div>
`;

exports[`should render info Badge 1`] = `
<div
  className="Badge__root Badge__info"
>
  2 new
</div>
`;

exports[`should render neutral Badge 1`] = `
<div
  className="Badge__root Badge__neutral"
>
  2 new
</div>
`;

exports[`should render secondary Badge 1`] = `
<div
  className="Badge__root Badge__secondary"
>
  2 new
</div>
`;
